### ISSUE TITLE
`http.server`: Update default `extensions_map` to include `.wasm`

### ISSUE THREAD

**Issue by vadimkantorov** [python/cpython#106900]
Then `python -m http.server` is suitable for experimenting with WASM/Emscripten projects. Currently people need to roll their own scripts like https://github.com/python/cpython/blob/main/Tools/wasm/wasm_webserver.py

Here are the default `extensions_map` (btw could include text/html, text/javascript, text/xml, text/json or other popular extensions for debugging very simple static web app projects): https://github.com/python/cpython/blob/a1a3193990cd6658c1fe859b88a2bc03971a16df/Lib/http/server.py#L661-L666

Adding MIME for `.data` by default is also nice for this is a file extension of Emscripten file system data packages.

It would also be nice if the default http server can be made multi-threaded, because in some rare circumstances it can block Emscripten data package loading. I rolled my own server https://gist.github.com/vadimkantorov/a8c3aca556824df061e84b1d10edee44 which also was serving ETag / HTTP 304 properly.

At least it would be nice to support customizing `extensions_map` in http.server command line, so that one can call `python -m http.server --map-mime .wasm:application/wasm --map-mime .data:application/octet-stream --serve-permanent-etag`

### Linked PRs
* gh-130430

**Comment by brettcannon** [comment]
I personally don't have an issue with adding `.wasm`, but I'm not sure about `.data` as that's extremely generic.

**Comment by vadimkantorov** [comment]
Yeah, I tend to agree about `.data` by default, but allowing the user to extend override/extend this list very simply via a few command-line options would be nice!

Some other two feature requests I proposed:
- check if it follows etags / allow to force browser caching (useful for debugging browser caching, e.g. Chrome won't cache anything larger than ~50mb which is a problem for large `.wasm` files for instance)
- allow to serve with multi-threading (otherwise sometimes Chrome hangs rendering in some rare situations)

**Comment by brettcannon** [comment]
> Some other two feature requests I'm proposing for the builtin `python -m http.server`

Those should be separate feature requests.

**Comment by vadimkantorov** [comment]
Sounds good! I'll factor out these two other feature requests in a separate issue.

Let's keep in this issue the discussion about proposal of also adding a command-line option for extending/overriding other MIME types like `python -m http.server --map-mime .wasm:application/wasm --map-mime .data:application/octet-stream` (or some similar command line syntax)?

**Comment by brettcannon** [comment]
> Let's keep in this issue the discussion about proposal of also adding a command-line option for extending/overriding other MIME types like `python -m http.server --map-mime .wasm:application/wasm --map-mime .data:application/octet-stream` (or some similar command line syntax)?

That's a separate request than adding `.wasm` to the default content-type mapping, so it should be a separate issue.

**Comment by vadimkantorov** [comment]
No problem, I can put it in separate issues for discussion! I agree, it is separate things :) Originally I stuffed more things in, but Alex renamed the issue :)

**Comment by donBarbos** [comment]
Looks like a good idea with no follow up so I'm going to send a PR

**Comment by freakboy3742** [comment]
Are we certain this problem actually exists? In my testing (on macOS, Python 3.14a5), a file named `foobar.wasm` is already returned as `application/wasm` because [that extension is registered as a default in `mimetypes`](https://github.com/python/cpython/blob/main/Lib/mimetypes.py#L508). The correct mimetype is also returned for `foobar.html`, `foobar.js`, `foobar.json`, `foobar.png`. This also appears to be the case back to [at least Python 3.10](https://github.com/python/cpython/blob/3.10/Lib/mimetypes.py#L459). 

While the WASM example server does need some custom code (the server has [moved to a new location](https://github.com/python/cpython/blob/main/Tools/wasm/emscripten/web_example/server.py) since this ticket was opened), that custom code doesn't need to register a mimetype - it's only needed to work around CORS limitations.

Am I missing something?

Command line registration of mimetypes is another issue entirely - that feature clearly doesn't exist... but given the list of default mimetypes, I'd have to assume the use cases for this are fairly obscure.

**Comment by donBarbos** [comment]
@freakboy3742 thanks for your comment. my mistake was that i didn't check the usecase, but it wasn't described here, it would be nice if someone interested in solving this issue showed an example. And you are right that wasm mime type was already added to `mimetypes.py`, but already 6 years ago (judging by git blame, [comment](https://github.com/python/cpython/commit/9fc720e5e4f772598013ea48a3f0d22b2b6b04fa#diff-aed43839a49bace08b60186baa4b27ad69ecd6b61f928bd696b4fb670750774f)) and this issue was opened 2 years ago.

**Comment by freakboy3742** [comment]
Ok - on that basis, I'm going close this wontfix. If I *have* missed something in my analysis, let me know, and we can re-open this.
